test/test_torch.py

```diff
@@ -3590,20 +3590,20 @@ def test_half_tensor(self):
             torch.save(xh, f)
             f.seek(0)
             xh2 = torch.load(f)
-            self.assertEqual(xh, xh2)
+            self.assertEqual(xh.float(), xh2.float())
 
     @unittest.skipIf(not torch.cuda.is_available(), 'no CUDA')
     def test_half_tensor_cuda(self):
         x = torch.randn(5, 5).half()
-        self.assertEqual(x.cuda().cpu(), x)
+        self.assertEqual(x.cuda(), x)
 
         xc = x.cuda()
         with tempfile.NamedTemporaryFile() as f:
             torch.save(xc, f)
             f.seek(0)
             xc2 = torch.load(f)
             self.assertIsInstance(xc2, type(xc))
-            self.assertEqual(xc, xc2)
+            self.assertEqual(xc.float(), xc2.float())
```

Comment:
Hmm @apaszke it looks like HalfTensor on CPU does not implement `ne` and hence failed tests.